### PR TITLE
[enterprise-4.14] OCPBUGS#33113: Doc improvements related to volume cloning/snapshots

### PR DESCRIPTION
**Pull request by sr1kar99**
Version(s):
4.14

Issue:
[OCPBUGS-33113](https://issues.redhat.com/browse/OCPBUGS-33113)

Link to docs preview:
[Preview](https://75330--ocpdocs-pr.netlify.app/openshift-enterprise/latest/storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.html#lvms-about-volume-snapsot_logical-volume-manager-storage)

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Comment by sr1kar99** [comment]
@jakobmoellerdev @suleymanakbas91 
Can you please review this PR?
Thanks!

**Comment by sr1kar99** [comment]
@radeore 
Can you please review this PR?
Thanks!

File: modules/lvms-creating-volume-clones.adoc [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+. Create the PVC in the namespace where you created the source PVC by running the following command:
++
+[source,terminal]
+----
+$ oc create -f <file_name> -n <namespace>
+----
+
+.Verification
+
+* To verify that the volume clone is created, run the following command:
```
Comment (radeore) [inline modules/lvms-creating-volume-clones.adoc:59]:
```suggestion
* To verify that the volume clone is created, first create a workload using created PVC and then run the following command:
```

File: modules/lvms-creating-volume-snapshots.adoc [hunk 2]
```diff
@@ -0,0 +1,75 @@
+// Module included in the following assemblies:
+//
+// storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="lvms-creating-volume-snapshots_{context}"]
+= Creating volume snapshots
+
+You can create volume snapshots based on the available capacity of the thin pool and the over-provisioning limits.
+To create a volume snapshot, you must create a `VolumeSnapshotClass` object.
```
Comment (radeore) [inline modules/lvms-creating-volume-snapshots.adoc:10]: `VolumeSnapshotClass` object always gets created by lvms for thinPool, I think we can skip this line saying 'you must create a VolumeSnapshotClass object'.

File: modules/lvms-restoring-volume-snapshots.adoc [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+. Create the PVC in the namespace where you created the the volume snapshot by running the following command:
++
+[source,terminal]
+----
+$ oc create -f <file_name> -n <namespace>
+----
+
+.Verification
+
+* To verify that the volume snapshot is restored, run the following command:
```
Comment (radeore) [inline modules/lvms-restoring-volume-snapshots.adoc:57]:
```suggestion
* To verify that the volume snapshot is restored, first create a workload using snapshot PVC and then run the following command:
```

**Comment by radeore** [comment]
Hi @sr1kar99  , added few comments, other then that LGTM. Thanks!

File: modules/lvms-creating-volume-snapshots.adoc (hunk 2 above)
Comment (sr1kar99) [reply modules/lvms-creating-volume-snapshots.adoc:10]: Agreed!
Here, I actually intended to write `VolumeSnapshot` object. Updated it to `VolumeSnapshot`.

File: modules/lvms-creating-volume-clones.adoc (hunk 1 above)
Comment (sr1kar99) [reply modules/lvms-creating-volume-clones.adoc:59]: Updated as per the suggestion.

File: modules/lvms-restoring-volume-snapshots.adoc (hunk 3 above)
Comment (sr1kar99) [reply modules/lvms-restoring-volume-snapshots.adoc:57]: Updated as per the suggestion.

File: storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc [hunk 4]
```diff
@@ -118,21 +118,22 @@ include::modules/lvms-scaling-storage-expand-pvc.adoc[leveloffset=+2]
...
-include::modules/lvms-creating-volume-snapshots-in-single-node-openshift.adoc[leveloffset=+2]
-include::modules/lvms-restoring-volume-snapshots-in-single-node-openshift.adoc[leveloffset=+2]
-include::modules/lvms-deleting-volume-snapshots-in-single-node-openshift.adoc[leveloffset=+2]
+include::modules/lvms-creating-volume-snapshots.adoc[leveloffset=+2]
+include::modules/lvms-restoring-volume-snapshots.adoc[leveloffset=+2]
+include::modules/lvms-deleting-volume-snapshots.adoc[leveloffset=+2]
 
 //Volume cloning
-include::modules/lvms-volume-clones-in-single-node-openshift.adoc[leveloffset=+1]
-include::modules/lvms-creating-volume-clones-in-single-node-openshift.adoc[leveloffset=+2]
```
Comment (adellape) [inline storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc:134]: FYI, there is one existing external `link` in the openshift-docs repo to this file on the Customer Portal, from a Microshift assembly:

https://github.com/openshift/openshift-docs/blob/main/microshift_storage/volume-snapshots-microshift.adoc?plain=1#L96

It won't 404 (it just dumps to the top of the page on the Portal), but you might want to update that `link:` to the new module location / URL. Possibly cross-check with a member of the Microshift docs team.

File: modules/lvms-about-volume-snapshots.adoc [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+[IMPORTANT]
+====
+Volume snapshots are located on the same devices as the original data. To use the volume snapshots as backups, you must move the snapshots to a secure location. You can use OpenShift API for Data Protection (OADP) backup and restore solutions. For information on OADP, see "OADP features".
+====
+
+* Revert to a state at which the volume snapshot was taken.
+
+[NOTE]
+====
+You can also create volume snapshots of the volume clones.
```
Comment (adellape) [inline modules/lvms-about-volume-snapshots.adoc:24]:
```suggestion
You can also create volume snapshots of volume clones. 
```

**Comment by adellape** [comment]
@sr1kar99 Can you confirm whether https://github.com/openshift/openshift-docs/pull/71446 is more ore less the 4.15+ PR for this similar content? To make it clear during merge-review why this is only targeting the `enterprise-4.14` branch. Thanks!

**Comment by sr1kar99** [comment]
> @sr1kar99 Can you confirm whether #71446 is more or less the 4.15+ equivalent PR for this similar content? To make it clear during merge-review why this is only targeting the `enterprise-4.14` branch. Thanks!

Hi @adellape 
Yes! You are right. Similar PR for 4.15+ has been merged already. The reason for having a separate PR for 4.14 is that 4.14 and lower versions support this Operator only on single-node OpenShift.

Thanks for adding the 4.15+ PR link. :)

File: storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc (hunk 4 above)
Comment (sr1kar99) [reply storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms.adoc:134]: This is a great catch! Thanks @adellape 
I've informed a writer from the MicroShift team. This link will be updated upon merging this PR.
Thanks!

File: modules/lvms-about-volume-snapshots.adoc (hunk 5 above)
Comment (sr1kar99) [reply modules/lvms-about-volume-snapshots.adoc]: Updated as per the suggestion.

**Pull request merged**